.github/actions/build-push-kotsadm-image/action.yml

```
@@ -0,0 +1,97 @@
+name: 'Build and push kotsadm image'
+description: 'Composite action for building and pushing kotsadm image'
+inputs:
+  chainguard-gcp-wif-pool:
+    description: 'GCP workload identity pool for Chainguard'
+    required: true
+
+  chainguard-gcp-sa:
+    description: 'GCP service account for Chainguard'
+    required: true
+
+  chainguard-gcp-project-id:
+    description: 'GCP project ID for Chainguard'
+    required: true
+
+  image-name:
+    description: 'Full destination kotsadm image name'
+    required: true
+
+  git-tag:
+    description: 'Git tag'
+    required: true
+
+  registry-username:
+    description: 'Username to login to registry'
+    default: ''
+    required: false
+
+  registry-password:
+    description: 'Password to login to registry'
+    default: ''
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - uses: google-github-actions/auth@35b0e87d162680511bf346c299f71c9c5c379033 # v1.1.1
+      with:
+        workload_identity_provider: ${{ inputs.chainguard-gcp-wif-pool }}
+        service_account: ${{ inputs.chainguard-gcp-sa }}
+
+    - uses: google-github-actions/setup-gcloud@e30db14379863a8c79331b04a9969f4c1e225e0b # v1.1.1
+      with:
+        project_id: ${{ inputs.chainguard-gcp-project-id }}
+
+    - name: setup packages gcsfuse
+      env:
+        BUCKET: replicated-apk-registry
+      shell: bash
+      run: |
+        # Install gcsfuse
+        export GCSFUSE_REPO=gcsfuse-`lsb_release -c -s`
+        echo "deb [signed-by=/usr/share/keyrings/cloud.google.asc] https://packages.cloud.google.com/apt $GCSFUSE_REPO main" | sudo tee /etc/apt/sources.list.d/gcsfuse.list
+        curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo tee /usr/share/keyrings/cloud.google.asc
+        sudo apt-get update -y
+        sudo apt-get install gcsfuse -y
+
+        # Set up a gcsfuse RO mount to the bucket containing private packages. This is a cheap and
+        # cheerful way to get access to objects we need, without having to fetch all of them.
+        mkdir -p /tmp/gcsfuse/apk-repo
+        gcsfuse -o ro --implicit-dirs --only-dir os ${BUCKET} /tmp/gcsfuse/apk-repo
+
+        # Symlink the gcsfuse mount to ./packages/$arch/*.apk
+        mkdir -p ./packages/x86_64
+        ln -s /tmp/gcsfuse/apk-repo/x86_64/*.apk ./packages/x86_64/
```

```suggestion
        ln -s /tmp/gcsfuse/apk-repo/x86_64/*.apk ./packages/x86_64/
        ln -s /tmp/gcsfuse/apk-repo/chainguard-enterprise.rsa.pub ./packages/
```

Also need to symlink the key into `./packages/`. We should be able to see this in the `ls -lR` below if this is successful.

---

Getting this error now (i only added the key to the melange config, not apko):
```
Error: failed to build package: unable to build guest: unable to create build context: initializing apk: failed to initialize apk keyring: failed to read apk key: open ./packages/chainguard-enterprise.rsa.pub: permission denied
2023/11/29 16:02:15 error during command execution: failed to build package: unable to build guest: unable to create build context: initializing apk: failed to initialize apk keyring: failed to read apk key: open ./packages/chainguard-enterprise.rsa.pub: permission denied
Error: Process completed with exit code 1.
```